extensions/positron-r/src/provider.ts

```diff
@@ -496,61 +378,234 @@ export async function findRBinaryFromPATHWindows(whichR: string): Promise<string
 	const binpathNormalized = firstExisting(whichRHome, binFragments());
 	if (binpathNormalized) {
 		LOGGER.info(`Resolved R binary at ${binpathNormalized}.`);
-		return binpathNormalized;
+		return { path: binpathNormalized, reasons: [ReasonDiscovered.PATH] };
 	} else {
 		LOGGER.info(`Can't find R binary within ${whichRHome}.`);
 		return undefined;
 	}
 }
 
-async function findRBinaryFromPATHNotWindows(whichR: string): Promise<string | undefined> {
+async function currentRBinaryFromPATHNotWindows(whichR: string): Promise<RBinary | undefined> {
 	const whichRCanonical = fs.realpathSync(whichR);
 	LOGGER.info(`Resolved R binary at ${whichRCanonical}`);
-	return whichRCanonical;
+	return { path: whichRCanonical, reasons: [ReasonDiscovered.PATH] };
+}
+
+function currentRBinaryFromHq(hqDirs: string[]): RBinary | undefined {
```

Comment:
This is actually a new check that applies only to non-Windows. I think it is very unlikely to surface different information that consulting `which`, but it's worth checking.